src/scraper/__init__.py

```diff
@@ -149,31 +298,76 @@ def scrape_main_page(self, base_url, odyssey_version, session, notes, logger, ms
                 ms_wait=ms_wait,
             )
             main_soup = BeautifulSoup(main_page_html, "html.parser")
-            return main_page_html, main_soup
+        except Exception as e:
+            logger.error(f"Error scraping main page for main page HTML: {e}")
```

Comment:
If you have an exception object when you log an error, prefer using this:
```suggestion
            logger.exception(e, "Error scraping main page for main page HTML")
```

This will print the stack trace and format the exception for you automatically

Reply:
Okay, this is great and I'll definitely implement this one in the next PR. Thanks!